src/Nethermind/Nethermind.Core/Utils/AutoCancelTokenSource.cs

```diff
@@ -0,0 +1,29 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Threading;
+
+namespace Nethermind.Core.Utils;
+
+/// <summary>
+/// Automatically cancel and dispose underlying cancellation token source.
+/// Make it easy to have golang style defer cancel pattern.
+/// </summary>
+public readonly struct AutoCancelTokenSource(CancellationTokenSource cancellationTokenSource) : IDisposable
```

Comment:
why not derive from CancellationTokenSource?

Reply:
I prefer not to use inheritance.

Reply:
I somewhat agree, but it IS a CancellationTokenSource, so it makes some sense